src/Orchestration/Adapter/DockerCLI.php

```diff
@@ -102,6 +102,135 @@ public function networkDisconnect(string $container, string $network, bool $forc
+     * @param array<string, string> $filters
+     * 
+     * @return array
+     */
+    public function getStats(string $container = null, array $filters = []): array
+    {
+        // List ahead of time, since docker stats does not allow filtering
+        $containerIds = [];
+
+        $stdout = '';
+        $stderr = '';
+
+        if($container === null) {
+            if(\count($filters) > 0) {
+                $filterString = '';
+
+                foreach($filters as $key => $value) {
+                    $filterString = $filterString . ' --filter "'.$key.'='.$value.'"';
+                }
+
+                $result = Console::execute('docker ps --all --no-trunc --format "id={{.ID}}"'.$filterString, '', $stdout, $stderr);
+
+                if($result !== 0) {
+                    throw new Orchestration("Docker Error: {$stderr}");
+                }
+
+                $stdoutArray = \explode("\n", $stdout);
+
+                foreach($stdoutArray as $row) {
+                    if(empty($row)) {
+                        continue;
+                    }
+
+                    $rowParsed = [];
+                    \parse_str($row, $rowParsed);
+                    $containerIds[] = $rowParsed['id'];
+                }
+            }
+        } else {
+            $containerIds[] = $container;
+        }
+
+        if(\count($containerIds) <= 0 && \count($filters) > 0) {
+            return []; // No containers found
+        }
+
+        $stats = [];
+
+        $containersString = "";
+
+        foreach ($containerIds as $containerId) {
+            $containersString .= " " . $containerId;   
+        }
+
+        $result = Console::execute('docker stats --no-trunc --format "id={{.ID}}&name={{.Name}}&cpu={{.CPUPerc}}&memory={{.MemPerc}}&diskIO={{.BlockIO}}&memoryIO={{.MemUsage}}&networkIO={{.NetIO}}" --no-stream' . $containersString, '', $stdout, $stderr);
+        
+        if($result !== 0) {
+            throw new Orchestration("Docker Error: {$stderr}");
+        }
+
+        $lines = \explode("\n", $stdout);
+
+        foreach($lines as $line) {
+            if(empty($line)) {
+                continue;
+            }
+
+            $stat = [];
+            \parse_str($line, $stat);
+
+            $stats[] = [
+                'id' => $stat['id'],
+                'name' => $stat['name'],
+                'cpu' => \floatval(\rtrim($stat['cpu'], '%')) / 100, // Remove percentage symbol, parse to number, convert to percentage
+                'memory' => empty($stat['memory']) ? 0 : \floatval(\rtrim($stat['memory'], '%')), // Remove percentage symbol and parse to number. Value is empty on Windows
+                'diskIO' => $this->parseIOStats($stat['diskIO']),
+                'memoryIO' => $this->parseIOStats($stat['memoryIO']),
+                'networkIO' => $this->parseIOStats($stat['networkIO']),
+            ];
+        }
+
+        return $stats;
+    }
+
+    private function parseIOStats(string $stats) {
```

Comment:
Can we have a doc block explaining this function?

Reply:
Added doc block.